osmhelper/osm_connector.py

```diff
@@ -0,0 +1,325 @@
+# coding=utf-8
+
+import sys
+import overpy
+from collections import OrderedDict
+from osmhelper.osm_routes import Route, RouteMaster
+from osmhelper.osm_stops import Stop
+from osmhelper.cache import Cache
+
+
+class OsmConnector(object):
+    """The OsmConnector class retrieves information about transit networks from
+    OpenStreetMap, caches it and serves it when needed to the osm2gtfs script
+
+    """
+
+    def __init__(self, config):
+        """Contructor function
+
+        This function gets called when OsmConnector object are created.
+
+        Based on the configuration from the config file it prepares the bbox
+        and tags that are used for querying OpenStreetMap.
+
+        Further it initializes main storage dictionary variables for routes and
+        stops, that are going to be shared and used by the ther components
+        of the program.
+
+        :param config: configuration information from the config file
+
+        """
+        self.config = config
+
+        # bbox from config file for querying
+        self.bbox = (str(config['query']['bbox']["s"]) + "," +
+                     str(config['query']['bbox']["w"]) + "," +
+                     str(config['query']['bbox']["n"]) + "," +
+                     str(config['query']['bbox']["e"]))
+
+        # tags from config file for querying
+        self.tags = ''
+        for key, value in config["query"].get("tags", {}).iteritems():
+            self.tags += str('["' + key + '" = "' + value + '"]')
+        else:
+            # fallback
+            self.tags = '["public_transport:version" = "2"]'
```

Comment:
Thanks! I noticed that the fenix config file actually has tags defined, but they are not used. There must be a bug here as well which might also cause some routes from missing that don't have the public_transport:version tag. Let's use the tags from the config file for now and change it later to public_transport:version.